### PR TITLE
TGALoader use DataTextureLoader

### PR DESCRIPTION
**Pull request by deepkolos**
**Description**

use DataTexture may use less resource then canvas

### CODE REVIEW

**Comment by Mugen87** [comment]
Intereseting idea! Do you mind modifing `example/js/loaders/TGALoader.js` instead and then generate the module via `node utils/modularize.js`(executed from the project's root directory).

**Comment by Mugen87** [comment]
Notice that the texture is now flipped:

https://raw.githack.com/deepkolos/three.js/master/examples/webgl_loader_texture_tga.html
https://threejs.org/examples/webgl_loader_texture_tga

When returning the texture data object in `parse()`, you have to set `flipY` to `true`. You also want to set `minFilter` to `LinearMipmapLinearFilter`.

**Comment by Mugen87** [comment]
To make this work, it's necessary to enhance `DataTextureLoader` so you can also set `generateMipmaps` to `true` in the result object`. Otherwise mipmaps are not generated.

**Comment by deepkolos** [comment]
thanks to point out my mistake.

switch back to Loader, and use DataTexture but reset to Texture's default setting

before

after

**Comment by Mugen87** [comment]
I would prefer the usage`DataTextureLoader`. I maps better to how other loaders work which produce a data texture.

**Comment by Mugen87** [comment]
Overriding `load()` should not be necessary if `DataTextureLoader` is enhanced at one place. But this can be refactored with a later PR.

**Comment by deepkolos** [comment]
if DataTextureLoader pass texture to the parse function, there is not need to Overriding load

```js
const texData = scope.parse( buffer, texture );
```

**Comment by Mugen87** [comment]
The return object in `parse()` should look like so:
```js
return {
    data: imageData,
    width: header.width,
    height: header.height,
    minFilter: THREE. LinearMipmapLinearFilter,
    generateMipmaps: true
};
```
`DataTextureLoader.parse()` needs this enhancement then:
```hs
if ( texData.generateMipmaps !== undefined ) {

    texture.generateMipmaps = texData.generateMipmaps;

}
```

File: src/loaders/DataTextureLoader.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ DataTextureLoader.prototype = Object.assign( Object.create( Loader.prototype ),
 		loader.setWithCredentials( scope.withCredentials );
 		loader.load( url, function ( buffer ) {
 
-			const texData = scope.parse( buffer );
+			const texData = scope.parse( buffer, texture );
```
Comment (Mugen87) [inline src/loaders/DataTextureLoader.js:35]: This change is non necessary.

File: build/three.js [hunk 2]
```diff
@@ -9106,17 +9106,17 @@
 		clone: function clone() {
 			/*
 			 // Handle primitives
-				 const parameters = this.parameters;
```
Comment (Mugen87) [inline build/three.js:9109]: Please do not include build files in the PR.

Comment (deepkolos) [reply build/three.js]: ok

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**